projects/plugins/jetpack/extensions/blocks/premium-content/edit.js

```diff
@@ -282,9 +287,19 @@ function Edit( props ) {
 		// Execution delayed with setTimeout to ensure it runs after any block auto-selection performed by inner blocks
 		// (such as the Recurring Payments block)
 		setTimeout( () => props.selectBlock(), 1000 );
-	// eslint-disable-next-line react-hooks/exhaustive-deps
+		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [] );
 
+	const shouldShowConnectButton = () => {
+		if ( ! shouldUpgrade && apiState !== API_STATE_CONNECTED && connectURL ) {
+			return true;
+		}
+
+		return false;
+	};
```

Comment:
Not a biggie, but this could be simplified to:

```
const shouldShowConnectButton = () => 
    ! shouldUpgrade && apiState !== API_STATE_CONNECTED && connectURL
```